Source/WebKit/GPUProcess/graphics/WebGPU/RemoteDevice.cpp

```diff
@@ -136,47 +136,51 @@ void RemoteDevice::destroy()
 
 void RemoteDevice::destruct()
 {
-    m_objectHeap->removeObject(m_identifier);
+    protectedObjectHeap()->removeObject(m_identifier);
 }
 
 void RemoteDevice::createXRBinding(WebGPUIdentifier identifier)
 {
+    auto objectHeap = protectedObjectHeap();
```

Comment:
We try to use smart pointers instead of `auto` to make it clearer when we're dealing with start pointers.